src/test/java/jenkins/plugins/horreum/junit/HorreumTestExtension.java

```diff
@@ -0,0 +1,138 @@
+package jenkins.plugins.horreum.junit;
+
+import com.cloudbees.plugins.credentials.Credentials;
+import com.cloudbees.plugins.credentials.CredentialsScope;
+import com.cloudbees.plugins.credentials.SystemCredentialsProvider;
+import com.cloudbees.plugins.credentials.domains.Domain;
+import com.cloudbees.plugins.credentials.impl.UsernamePasswordCredentialsImpl;
+import io.hyperfoil.tools.horreum.api.services.ConfigService;
+import io.hyperfoil.tools.horreum.infra.common.SelfSignedCert;
+import jenkins.plugins.horreum.HorreumGlobalConfig;
+import jenkins.plugins.horreum.JenkinsExtension;
+import org.jboss.logging.Logger;
+import org.junit.jupiter.api.extension.*;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static io.hyperfoil.tools.horreum.infra.common.Const.*;
+import static io.hyperfoil.tools.horreum.infra.common.HorreumResources.startContainers;
+import static java.lang.System.getProperty;
+import static jenkins.plugins.horreum.it.JenkinsResources.*;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+public class HorreumTestExtension {
+
+    private static final Logger log = Logger.getLogger(HorreumTestExtension.class);
+    private static boolean started = false;
+    public static String horreumHost;
+    public static String horreumPort = "8080";
+    private static final Map<Domain, List<Credentials>> credentials = new HashMap<>();
+    public static final String HORREUM_UPLOAD_CREDENTIALS = "horreum-creds";
+    public static final String DEFAULT_BOOTSTRAP_USERNAME = "horreum.bootstrap";
+    public static final String DEFAULT_BOOTSTRAP_PASSWORD = "secret";
+
+    @RegisterExtension
+    public static JenkinsExtension j = new JenkinsExtension();
+
+    public static void beforeAll() throws Exception {
+        synchronized (HorreumTestExtension.class) {
+            if (!started) {
+                log.info("Starting Jenkins IT resources");
+
+                try {
+                    String keycloakImage = getProperty(HORREUM_DEV_KEYCLOAK_IMAGE);
+                    String postgresImage = getProperty(HORREUM_DEV_POSTGRES_IMAGE);
+                    String amqpImage = getProperty(HORREUM_DEV_AMQP_IMAGE);
+                    String horreumImage = getProperty(HORREUM_DEV_HORREUM_HORREUM_IMAGE);
```

Comment:
this should be loaded from `META-INF/horreum.dev.properties` published in `horreum-infra-common` module, after this PR has been merged: https://github.com/Hyperfoil/Horreum/pull/2023

Reply:
updated to load from the properties file.